src/course-search/FacetDropdown.jsx

```diff
@@ -1,33 +1,79 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Dropdown } from '@edx/paragon';
+import { Dropdown, Input } from '@edx/paragon';
 import classNames from 'classnames';
 
-const FacetDropdown = ({ title, items, isBold }) => (
-  <div className="facet-list">
-    <Dropdown className="mb-0 mr-md-3">
-      <Dropdown.Toggle
-        className={
-          classNames(
-            'bg-white', 'text-capitalize', 'rounded-0', 'border-0',
-            'd-flex', 'justify-content-between', 'align-items-center', 'text-dark',
-            { 'font-weight-bold': isBold },
-          )
-        }
-      >
-        {title}
-      </Dropdown.Toggle>
-      <Dropdown.Menu>
-        {items}
-      </Dropdown.Menu>
-    </Dropdown>
-  </div>
-);
+const FacetDropdown = ({
+  title,
+  items,
+  isBold,
+  typeahead,
+  searchForItems,
+}) => {
+  const handleSearch = (value) => {
+    // when user is erasing the input and input is empty we need to reset the filtering
+    if (value.length >= typeahead.minLength || value.length === 0) {
+      searchForItems(value);
+    }
+  };
+
+  return (
+    <div className="facet-list">
+      <Dropdown className={classNames('mb-0 mr-md-3', { typeahead: !!typeahead })}>
+        <Dropdown.Toggle
+          className={
+            classNames(
+              'bg-white', 'text-capitalize', 'rounded-0', 'border-0',
+              'd-flex', 'justify-content-between', 'align-items-center', 'text-dark',
+              { 'font-weight-bold': isBold },
+            )
+          }
+        >
+          {title}
+        </Dropdown.Toggle>
+        <Dropdown.Menu>
+          {
+            typeahead && (
+              <div className="search-input-wrapper">
+                <Input
+                  autoFocus
+                  type="search"
+                  className="dropdown-typeahead"
+                  placeholder={typeahead.placeholder}
+                  aria-label={typeahead.ariaLabel}
+                  onChange={(event) => handleSearch(event.currentTarget.value)}
+                />
+              </div>
+            )
+          }
+          {
+            typeahead ? (
+              <div className="dropdown-menu-scrollable-items">
```

Comment:
We should test that the correct items list appears based on whether or not the typeahead is present.

Reply:
This may be easier to test if the inner part of the component is split into a new component.